NetTopologySuite.IO/NetTopologySuite.IO/IO/Streams/FileStreamProvider.cs

```diff
@@ -19,10 +21,10 @@ public FileStreamProvider(string kind, string path, bool validatePath = false)
             if (path == null)
                 throw new ArgumentNullException("path");
 
-#if NET40
+#if HAS_SYSTEM_STRING_ISNULLORWHITESPACE
             if (string.IsNullOrWhiteSpace(path))
 #else
-            if (string.IsNullOrEmpty(path))
+            if (string.IsNullOrEmpty(path) || path.All(Char.IsWhiteSpace))

```

Comment:
>  I expect it to terminate after reading exactly one character

so the same is valid for LINQ. personally, I'd like this linq functions, but feel free to use what do you like.